packages/aws-cdk-lib/aws-route53/lib/geo-location.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Routing based on geographical location.
+ */
+export class GeoLocation {
+  /**
+   * Geolocation resource record based on continent code.
+   * @param continentCode Code of the continent. Valid codes are AF (Africa), AN (Antarctica), AS (Asia),
+   * EU (Europe), OC (Oceania), NA (North America), SA (South America).
+   * @returns Continent-based geolocation record
+   */
+  public static continent(continentCode: string) {
+    return new GeoLocation(continentCode, undefined, undefined);
+  }
+
+  /**
+   * Geolocation resource record based on country code.
+   * @param countryCode Two-letter, uppercase country code for the country.
+   * See ISO 3166-1-alpha-2 code on the *International Organization for Standardization* website
+   * @returns Country-based geolocation record
+   */
+  public static country(countryCode: string) {
+    return new GeoLocation(undefined, countryCode, undefined);
+  }
+
+  /**
+   * Geolocation resource record based on subdivision code (e.g. state of the United States).
+   * @param subdivisionCode Code of the subdivision (e.g. state of the United States)
+   * @param countryCode Country code (ISO 3166-1-alpha-2) of this record, by default US (United States).
+   * @returns Subdivision-based geolocation record
+   */
+  public static subdivision(subdivisionCode: string, countryCode: string = 'US') {
+    return new GeoLocation(undefined, countryCode, subdivisionCode);
+  }
```

Comment:
CloudFormation (and the Route 53 API) _forbid_ specifying `subdivision` when the country code is value other than US. At least for now, would it better to add a check for that to catch it at synthesis time (though maybe in the constructor instead of right here)?


```ts
route53.GeoLocation.subdivision('XX', 'CA'); // -> "Subdivisions are only supported when the countryCode is 'US'"
```

> For geolocation resource record sets, the two-letter code for a state of the United States. Route 53 doesn't support any other values for `SubdivisionCode`. 
> If you specify `subdivisioncode`, you must also specify `US` for `CountryCode`.
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-route53-recordset-geolocation-1.html#cfn-route53-recordset-geolocation-subdivisioncode

Reply:
Route53 also supports `subdivisioncode` for `CountryCode` US (Ukraine). However, I can't find it in the documentation, too. It can be maintained using AWS console and CloudFormation. AWS console:
<img width="351" alt="image" src="https://github.com/aws/aws-cdk/assets/15660169/61b05410-f9e7-4209-aed8-ab6b9f6ae879">

I added an integration test for a subdivision in Ukraine which can be deployed successfully.
```ts
new ARecord(stack, 'GeoLocationSubDividionUA', {
  zone: privateZone,
  target: RecordTarget.fromIpAddresses('1.2.3.4', '5.6.7.4'),
  geoLocation: GeoLocation.subdivision('30', 'UA'), // Ukraine, Kyiv
});
```

Which field order do you prefer?
Option 1: SubDivision, Country (country can be optional for US).
```ts
GeoLocation.subdivision('WA', 'US')
GeoLocation.subdivision('30', 'UA')
```

Option 2: Country, SubDivision (country 
```ts
GeoLocation.subdivision('US', 'WA')
GeoLocation.subdivision('UA', '30')
```

I think I would prefer option 2 because the generic information (country) has to be specified first.

Reply:
I added validation for country code in subdivision, only UA (Ukraine) and US (United States) is allowed now.

Reply:
@colifran Can you also give a feedback to this questions? Should I implement option 2 (as proposed)? Or do you prefer option 1? Thanks.

Reply:
@jumic My preference would be option 1. Since the static method is `subdivision`, I think it makes more sense for the first field to be the `subdivision` to use followed by the country that the `subdivision` is for.

Reply:
OK, then we will use option 1. In this case, no further changes are necessary.